### PR TITLE
Fix parsing of expires_on

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Handle expires_on time stamps that omit an AM/PM designation.

Thank you for your contribution to Go-AutoRest! We will triage and review it as soon as we can.

As part of submitting, please make sure you can make the following assertions:
 - [ ] I've tested my changes, adding unit tests if applicable.
 - [ ] I've added Apache 2.0 Headers to the top of any new source files.
 - [ ] I'm submitting this PR to the `dev` branch, except in the case of urgent bug fixes warranting their own release.
 - [ ] If I'm targeting `master`, I've updated [CHANGELOG.md](https://github.com/Azure/go-autorest/blob/master/CHANGELOG.md) to address the changes I'm making.

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
Fixes https://github.com/Azure/go-autorest/issues/593

**Review by catalinaperalta — APPROVED** [review]
lgtm

File: autorest/adal/token.go [hunk 1]
```diff
@@ -1024,13 +1027,18 @@ func (spt *ServicePrincipalToken) refreshInternal(ctx context.Context, resource
 
 // converts expires_on to the number of seconds
 func parseExpiresOn(s string) (json.Number, error) {
+	// convert the expiration date to the number of seconds from now
+	timeToDuration := func(t time.Time) json.Number {
+		dur := t.Sub(time.Now().UTC())
+		return json.Number(strconv.FormatInt(int64(dur.Round(time.Second).Seconds()), 10))
+	}
```
Comment (mdgreenfield) [inline autorest/adal/token.go:1034]: @jhendrixMSFT, I'm a little confused why this calculates the difference between `t time.Time` (the passed in and parsed `s string`) and `time.Now().UTC()`. That difference I believe is effectively the `expires_in` value (the TTL of the token). 

However, when the `strconv.ParseInt(s, 10, 64)` logic is successful this `parseExpiresOn` function would be passing through the unix timestamp as far as I can tell (i.e. the actual time when the token expires and not the `expires_in` value).

From the IMDS docs - https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/how-to-use-vm-token#get-a-token-using-http

> `expires_on` - The timespan when the access token expires. The date is represented as the number of seconds from "1970-01-01T0:0:0Z UTC" (corresponds to the token's `exp` claim).

Comment (mdgreenfield) [reply autorest/adal/token.go:1034]: I've moved this question to https://github.com/Azure/go-autorest/issues/638. Apologies for the noise here.

**Pull request merged**